<launch>

<!-- Default folders with maps-->

<arg name="folder_map" default="$(env HOME)/hosti"/>
<arg name="folder_view" default="$(env HOME)/hosti"/>

<!-- Distance Measurement Node -->
<node pkg="stroll_bearnav" type="odometry_monitor_node" name="odometry_monitor"/>

<!-- feature extraction -->
  <node pkg="stroll_bearnav" type="feature_extraction_node" name="feature_extraction">
	  <remap from="image" to="/stereo/left/image_raw"/>
  </node>

<!-- mapping node -->
<node pkg="stroll_bearnav" type="mapper_node" name="mapper" output="screen">
  <remap from="image" to="/stereo/left/image_raw"/>
  <param name="folder" value="$(arg folder_map)" />
  <param name="angularSpeed" value="1.1" />
</node>

<!-- map loader1 node -->
<node pkg="stroll_bearnav" type="map_preprocessor_node" name="map_preprocessor_map" output="screen">
<param name="folder" value="$(arg folder_map)" />
<remap from="map_preprocessor" to="map_preprocessor_map"/>
</node>

<!-- navigator node -->
<node pkg="stroll_bearnav" type="navigator_node" name="navigator" output="screen">
  <param name="folder" value="$(arg folder_map)" />
  <!--remap from="image_with_features" to="image_view"/-->
</node>

<!-- navigator node -->
<node pkg="stroll_bearnav" type="auto_remapper" name="auto_remapper" output="screen">
  <param name="folder_map" value="$(arg folder_map)" />
  <param name="folder_view" value="$(arg folder_view)" />
</node>

<!-- action and reconfigure clients to control the system -->
<!--node pkg="rqt_reconfigure" type="rqt_reconfigure" name="reconfigurator"/-->

<!-- image view to check video stream-->
<node pkg="image_view" type="image_view" name="image_viewer">
  <!--param name="~image_transport" value="compressed" /-->
  <remap from="image" to="/navigationMatches"/>
  <param name="autosize" value="true" />
</node>

<!-- image view to check video stream-->
<node pkg="image_view" type="image_saver" name="image_saver">
  <remap from="image" to="/navigationMatches"/>
  <param name="autosize" value="true" />
</node>




</launch>
